documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

```diff
@@ -1400,6 +1400,10 @@ include::{testDir}/example/ParameterizedTestDemo.java[tags=ArgumentsSource_examp
 include::{testDir}/example/ParameterizedTestDemo.java[tags=ArgumentsProvider_example]
 ----
 
+NOTE: Arguments of type `AutoCloseable` will be closed once `@AfterEach` methods and
+`AfterEachCallback` implementations have been called. To prevent this from happening, set
+the `autoCloseArguments` attribute in `@ParameterizedTest` to `false`.
+
```

Comment:
I just realized this note is in the `@ArgumentSource` section; whereas, it applies to all arguments from any _source_.

I know that all other `@*Source` annotations are meta-annotated with `@ArgumentSource`, but the average user doesn't.

So I think we should add this note higher up in the chapter -- for example, in the _Consuming Arguments_ section.

@juliette-derancourt / @marcphilipp, thoughts?

Reply:
I put it here as "at the end of the whole _Sources of Arguments_ section" more than in the `@ArgumentSource` section particularly, but you're right it might be better the _Consuming Arguments_ section.